### DISCUSSION TITLE
API reference for Multi Label Graphs

### DISCUSSION THREAD

**Discussion by sreekarreddy21** [networkx/networkx#6162] [Q&A] [answered]
Hi,

I am trying to find the API reference to generate graphs in case of multilabel data. For Instance, I have a dataset of 1000 nodes, belonging to 4 groups. I want to study the interaction between these 1000 nodes of multiple groups by generating graphs on this data. I looked at multilayered disjoint graphs, I couldn't find the related API reference in networkx? How do we generate such graphs using networkx? Could anyone please point me to the right reference?

Thanks.

**Comment by dschult** [comment, marked as answer]
It is not clear what you mean. There are many ways to construct graphs with 1000 nodes each of which is an element of 4 groups. One way is to create a graph with 1000 nodes and then make a node attribute to indicate which group they are in:
```python
for n in G:
    G.nodes[n]['group']=random.randint(0,3)
```

If you start with graphs for each group and you want to join them into a single group try [disjoint_union](https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.operators.binary.disjoint_union.html)

**Reply by sreekarreddy21** [reply]
Great, I will try this method and let you know if I have any further questions. Thanks.
